zmon-security-common/src/main/java/de/zalando/zmon/security/tvtoken/TvTokenService.java

```diff
@@ -0,0 +1,41 @@
+package de.zalando.zmon.security.tvtoken;
+
+import org.springframework.util.Base64Utils;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+import de.zalando.zmon.persistence.OnetimeTokensSProcService;
+
+public class TvTokenService {
+
+    private final OnetimeTokensSProcService oneTimeTokenSProcService;
+
+    private final Joiner cookieValueJoiner = Joiner.on("|").useForNull("UNKNOWN");
+    private final Splitter cookieValueSplitter = Splitter.on("|").omitEmptyStrings();
+
+    public TvTokenService(OnetimeTokensSProcService onetimeTokensSProcService) {
+        this.oneTimeTokenSProcService = onetimeTokensSProcService;
+    }
+
+    public String createCookieValue(String token, String bindIp, String sessionId) {
+        String cookieValue = cookieValueJoiner.join(token, bindIp, sessionId);
+        String encodedValue = Base64Utils.encodeToString(cookieValue.getBytes());
+        return encodedValue;
+    }
+
+    public Iterable<String> decodeCookieValue(String encodedCookieValue) {
+        String decodedValue = new String(Base64Utils.decodeFromString(encodedCookieValue));
+        return cookieValueSplitter.split(decodedValue);
+    }
+
+    // TODO use sproc
+    public boolean isValidToken(String token, String bindIp, String sessionId) {
```

Comment:
That sounds good.

Yes "session id" is not in general used, but the token alone is too short to use, so we need another secret that is only known to user's browser. So assuming session id is hard to guess which it should be, that is our safety (token, session, ip). That is the only reason for binding to session id.